### PR TITLE
Milestone 1

### PR DESCRIPTION
**Pull request by Trictonicmp**
#### In this PR I:

- [x]  Create class `Person` with  the following:

  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.

- [x]  Create class `Student` with the following:

  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.

- [x]  Create class `Teacher` with the following:

  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- [x] Each class should be saved in a separate file.

### CODE REVIEW

File: classes/person.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = 0
```
Comment (Davidosky007) [inline classes/person.rb:12]: * Kindly create a getter method for `@id` like you did for `@age` and `@name` as stated in [the requirement ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Getters%20for%20%40id%2C%20%40name%2C%20and%20%40age.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Trictonicmp 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written readme file 👍 
* Good looking PR description 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb (hunk 1 above)
Comment (Trictonicmp) [reply classes/person.rb:12]: Hey @Davidosky007 Thank you for the feedback, I made the changes as requested.

**Review by gemmen29 — APPROVED** [review]
Hi @Trictonicmp, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**